tests/test_integration.py

```diff
@@ -0,0 +1,57 @@
+# SPDX-License-Identifier: MIT
+
+import os
+import os.path
+import re
+
+import pytest
+
+import build.__main__
+
+
+_SDIST = re.compile('.*.tar.gz')
+_WHEEL = re.compile('.*.whl')
+
+
+@pytest.mark.parametrize(
+    ('project'),
+    [
+        'python-build',
+        'pip',
+        'dateutil',
+        'Solaar',
+    ]
+)
+@pytest.mark.parametrize(
+    ('args'),
+    [
+        [],
+        ['-x', '--no-isolation'],
+    ]
+)
+def test_build(tmpdir, integration_path, project, args):
+    os.environ['SETUPTOOLS_SCM_PRETEND_VERSION'] = 'dummy'  # for the projects that use setuptools_scm
```

Comment:
use pytests monkeypatch.setenv instead of atlering your os environ. This is needed to ensure that subsequent tests are not affected by changes to a global state in this test.